### PR TITLE
Prod 146: Design freindly errors

### PR DESCRIPTION
**Pull request by yash431garg**
- Description
Handle runtime errors with error pages.
- What are the steps to test that this code is working?
1. 404 page on wrong page route
2. 400 client error page on client exception
3. 500 server error page on server exception
- Screen shots or recordings for UI changes
Screenshot 2024-09-13 at 12 25 48 PM

### CODE REVIEW

**Comment by yash431garg** [comment]
@Yihwan Can you please review the desings here?

File: app/components/Icons/RefreshIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ICON_DEFAULT_WIDTH_HEIGHT, IconProps } from ".";
```
Comment (Yihwan) [inline app/components/Icons/RefreshIcon.tsx:1]: we no longer need to import icons manually. pls try importing from https://lucide.dev/guide/packages/lucide-react and see if it is easy to get it to look the same as before; if not we can address in separate PR

File: app/not-found.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        </div>
+      </div>
+
+      <div className="flex flex-col items-start text-white mt-2 gap-2">
+        <p className="text-[16px] mb-2">
+          Page not found. Let&apos; s go somewhere else!
+        </p>
+        <p className="text-[14px] mb-2">Optional error description</p>
+        <p className="text-[14px]">
+          (stack trace, other technical info that may be helpful for debugging)
```
Comment (Yihwan) [inline app/not-found.tsx:32]: pls remove all placecholder copy. it is helpful for checking design but if no actual copy was provided pls just remove

**Review by Yihwan — CHANGES_REQUESTED** [review]
icon changes are optional if they require more discussion, but we have to remove all placeholder copy before this can be merged

File: app/not-found.tsx (hunk 2 above)
Comment (yash431garg) [reply app/not-found.tsx:32]: Sure, I will remove this.

File: app/components/Icons/RefreshIcon.tsx (hunk 1 above)
Comment (yash431garg) [reply app/components/Icons/RefreshIcon.tsx:1]: I have changed using lucide.

File: app/components/ErrorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import Image from "next/image";
+import Link from "next/link";
+import { Button } from "../ui/button";
+import { RefreshCw, ArrowLeft } from 'lucide-react';
+
+function ErrorBoundary({ error, reset }: { error: Error; reset: () => void }) {
+  const router = useRouter();
+
+  useEffect(() => {
+    Sentry.captureException(error);
```
Comment (marvinmarnold) [inline app/components/ErrorBoundary/ErrorBoundary.tsx:13]: Let's use a custom error type for this so it's easier to track. Like this: https://github.com/gator-labs/chomp/blob/1f14fbbf2335392fff2c3d18af35b3443caa58d3/app/providers/DynamicProvider.tsx#L64

You can call it ErrorBoundaryError or something

**Review by marvinmarnold — CHANGES_REQUESTED** [review]

**Pull request merged**